### ISSUE TITLE
citations in 'level two' term cause error when both are not used.

### ISSUE THREAD

**Issue by TomVer99** [typst-community/glossarium#55]
When you have two terms, `A` and `B`. And term `A` uses term `B` in its `desc`, and term `B` has a citation. But both terms are not used in the document, you get the following error:

<img width="340" alt="Code_dV7CTTNVpg" src="https://github.com/user-attachments/assets/1af05bc6-c103-429c-a595-561e22779b93">

If either of the two terms is used, it works again. But if term `B`, again uses term `C`, and term `C` cites a source, the same error occurs.

**Comment by TomVer99** [comment]
This issue also seems to have to do with using terms in terms in terms

as soon as you are '3 deep' things break.

**Comment by quachpas** [comment]
Hello @TomVer99 , would you be able to provide a minimal working example? Thanks!

**Comment by TomVer99** [comment]
> Hello @TomVer99 , would you be able to provide a minimal working example? Thanks!

Do you mean a typst file where the error occurs?

**Comment by quachpas** [comment]
A document that reproduces this specific situation yes!

**Comment by TomVer99** [comment]
@quachpas I think this demonstrates the problem. The fourth term does not get printed although it does gets cited by the third. If you cite the second term, the fourth term does appear.

```typst
#import "@preview/glossarium:0.4.1": make-glossary, print-glossary, gls, glspl
#show: make-glossary

#let glossary-terms = (
  (key: "unamura", short: "UNamura", long: "Namur Universitya", desc: [test a @unamurb]),
  (key: "unamurb", short: "UNamurb", long: "Namur Universityb", desc: [test b @unamurc]),
  (key: "unamurc", short: "UNamurc", long: "Namur Universityc", desc: [test c @unamurd]),
  (key: "unamurd", short: "UNamurd", long: "Namur Universityd", desc: [test d]),
)

@unamura sdsadas

// @unamurb sdsadas

#print-glossary((glossary-terms))
```

**Comment by quachpas** [comment]
Hello, these are the two outputs I get with your code, thanks!
Without @b
![image](https://github.com/user-attachments/assets/29d9564f-0e1e-4c1f-87d6-a5001b7d2b89)
With 
![image](https://github.com/user-attachments/assets/7545ff2e-0382-4646-87ed-720c454b6199)

The problem is that with 0.4.1, we are using `final` and the layout cannot converge. 
```shell
warning: layout did not converge within 5 attempts
 = hint: check if any states or queries are updating themselves
```
this is the error you should have gotten during compilation.

Unfortunately, this behaviour cannot be supported in 0.4.1. 

This is fixed in version 0.4.2, only references in the main document count towards showing references.
![image](https://github.com/user-attachments/assets/f7d7aa43-bb3f-40ac-bd5e-4330695c2032)

**Comment by quachpas** [comment]
@TomVer99, just merged #100 that should enable infinite recursion (for better or worse)

**Comment by TomVer99** [comment]
> @TomVer99, just merged #100 that should enable infinite recursion (for better or worse)

To infinity! And beyond!!!!!!

**Comment by quachpas** [comment]
hmmmm I think I spoke too quickly. I was using `show-all: true`.
